src/index.ts

```diff
@@ -68,17 +68,22 @@ function comparadorEdadYAltura(edad: number[], altura: number[], i: number) {
 }
 
 //Intercambio los datos del arreglo moviendolo 1 posicion, guardo el dato a mover en copia
-function intercambiarValores(arreglo: number[], i: number) {
+function intercambiarValores(arreglo: any[], i: number) {
   let copia: number;
   let b: number = i + 1;
   copia = arreglo[i];
   arreglo[i] = arreglo[b];
   arreglo[b] = copia;
 }
 
-function burbuja(nombres: string[], edad: number[], altura: number[]) {
-  for (let i = 0; i < nombres.length; i++) {
-    for (let j = 0; j < nombres.length - 1 - i; j++) {
+function burbuja(
+  nombres: string[],
+  edad: number[],
+  altura: number[],
+  dimensionArray: number
+) {
+  for (let i = 0; i < dimensionArray - 1; i++) {
+    for (let j = 0; j < dimensionArray - 1 - i; j++) {
```

Comment:
Al **for** de **i** hacemos que vaya hasta **dimensionArray - 1**, faltaba el -1.